src/prefect/_internal/concurrency/runtime.py

```diff
@@ -96,17 +96,16 @@ def __init__(self, name: str = "RuntimeLoopThread"):
         self._worker_processes = Executor(
             worker_type="process", initializer=_initialize_worker_process
         )
-        self._ready_event = threading.Event()
+        self._ready_future = concurrent.futures.Future()
         self._loop: Optional[asyncio.AbstractEventLoop] = None
 
     def start(self):
         """
-        Start the thread and wait until the event loop is ready.
+        Extends the default `Thread.start()` to wait until `run` is ready and reraise
+        any errors encountered during setup.
         """
         super().start()
-        # Extends the typical thread start event to include a readiness event set from
-        # within the event loop.
-        self._ready_event.wait()
+        self._ready_future.result()
```

Comment:
Here we use a future instead of an event to raise errors that occur during startup of the thread.